### PR TITLE
First revision of alarming

### PR DESCRIPTION
**Pull request by mcnamarabrian**

### CODE REVIEW

File: template.yml [hunk 1]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      TreatMissingData: notBreaching
+      Dimensions:
+        - Name: QueueName
+          Value:
+            Ref: ConversionDlq
+
+  SentimentDlqAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Alarm if Sentiment DLQ queue depth is more than 1 message
```
Comment (jrrudge) [inline template.yml]: I think this should be more than 0 messages

File: template.yml [hunk 2]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      AlarmActions:
+        - Ref: AlarmTopic
+      MetricName: ApproximateNumberOfMessagesVisible
+      Namespace: AWS/SQS
+      Statistic: Sum
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 1
+      ComparisonOperator: GreaterThanOrEqualToThreshold
```
Comment (jrrudge) [inline template.yml]: GreaterThan

File: template.yml [hunk 3]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+    Properties:
+      AlarmDescription: Alarm if Sentiment DLQ queue depth is more than 1 message
+      AlarmActions:
+        - Ref: AlarmTopic
+      MetricName: ApproximateNumberOfMessagesVisible
+      Namespace: AWS/SQS
+      Statistic: Sum
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 1
```
Comment (jrrudge) [inline template.yml]: Threshold 0

File: template.yml [hunk 4]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+      Dimensions:
+        - Name: QueueName
+          Value:
+            Ref: ConversionDlq
```
Comment (jrrudge) [inline template.yml]: Ref: returns the URL, This should be Fn::GetAtt: [ConversionDlq, QueueName]

File: template.yml [hunk 5]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+    Type: AWS::SNS::Topic
+    Properties:
+      Subscription:
+        - Protocol: email
+          Endpoint: !Ref AlarmRecipientEmailAddress
+
+  ConversionDlqAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Alarm if Conversion DLQ queue depth is more than 1 message
```
Comment (jrrudge) [inline template.yml]: Should read depth is greater than zero messages, alternatively 1 or more

File: template.yml [hunk 6]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      AlarmActions:
+        - Ref: AlarmTopic
+      MetricName: ApproximateNumberOfMessagesVisible
+      Namespace: AWS/SQS
+      Statistic: Sum
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 1
```
Comment (jrrudge) [inline template.yml]: Threshold 0

File: template.yml [hunk 7]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      ComparisonOperator: GreaterThanOrEqualToThreshold
```
Comment (jrrudge) [inline template.yml]: GreaterThan

File: template.yml [hunk 8]
```diff
@@ -272,3 +277,48 @@ Resources:
...
+      TreatMissingData: notBreaching
+      Dimensions:
+        - Name: QueueName
+          Value:
+            Ref: SentimentDlq
```
Comment (jrrudge) [inline template.yml]: Ref: returns the URL, This should be Fn::GetAtt: [ConversionDlq, QueueName]

**Review by jrrudge — COMMENTED** [review]
Made a couple of comments. It works for me if you change the config as per the comments.

**Review by jrrudge — APPROVED** [review]
All Working. Have some Tacos 🌮 🌮 🌮 🌮

**Comment by mcnamarabrian** [comment]
Thanks much, @jrrudge.  I'll enjoy those 🌮 - you do the same!

**Pull request merged**